tests/test_url.py

```diff
@@ -1240,6 +1240,23 @@ def test_with_path_fragment():
     assert str(url.with_path("/test")) == "http://example.com/test"
 
 
+@pytest.mark.parametrize(
+    "original_url, keep_query, keep_fragment, expected_url",
+    [
+        ("http://example.com?a=b#frag", True, False, "http://example.com/test?a=b"),
+        ("http://example.com?a=b#frag", False, True, "http://example.com/test#frag"),
+        ("http://example.com?a=b#frag", True, True, "http://example.com/test?a=b#frag"),
+        ("http://example.com?a=b#frag", False, False, "http://example.com/test"),
+    ],
```

Comment:
Let's add context by naming each param:
```suggestion
    ("original_url", "keep_query", "keep_fragment", "expected_url"),
    (
        pytest.param("http://example.com?a=b#frag", True, False, "http://example.com/test?a=b", id="query-only"),
        pytest.param("http://example.com?a=b#frag", False, True, "http://example.com/test#frag", id="fragment-only"),
        pytest.param("http://example.com?a=b#frag", True, True, "http://example.com/test?a=b#frag", id="all"),
        pytest.param("http://example.com?a=b#frag", False, False, "http://example.com/test", id="none"),
    ),
```

Reply:
Agree, the proposed change makes tests much better readable.